docs/proposals/29-autoscaling-storage-volumes.md

```diff
@@ -0,0 +1,335 @@
+Its primary driving signal is the PVC metrics from the seed's cache Prometheus instance, which it
+periodically examines, and if capacity is found to be near exhaustion, `pvc-autoscaler` takes action by updating the
+PVC's storage request.
+
+In this initial iteration, `pvc-autoscaler` scales two categories of `vali` and `prometheus` volumes:
+those in shoot namespaces, and those in the seed's `garden` namespace.
+
+`pvc-autoscaler` publishes Prometheus metrics to anonymous scrapers via a `/metrics` HTTP endpoint. A `ServiceMonitor`
+object is created in the `garden` namespace, and drives the seed's `prometheus-operator` to configure a scrape on that
+endpoint by `prometheus-seed`.
+
+`pvc-autoscaler` runs in active/passive replication mode, based on the standard leader election mechanism, supplied by
+the K8s controller runtime library. Only the active replica is shown in _Fig.1_.
+
+### PVC Autoscaler - Deployer Structure
+The `pvc-autoscaler` application and its supporting artefacts are deployed by `gardener`, as part of the seed
+reconciliation flow.
+
+The `vali` and `prometheus` deployers, part of `gardenlet`'s shoot reconciliation flow, ensure that the PVC template section in
+the respective StatefulSet objects contains the annotations, necessary to instruct `pvc-autoscaler` how to act on the
+PVCs, created from those templates.
+
+That `gardenlet` action mechanism, based on PVC templates, only affects PVCs upon creation. Therefore, it has no effect
+on existing shoots, and on the observability PVCs of existing seed clusters. To enable PVC autoscaling for existing
+shoots, those PVCs will be directly annotated by the operator. Such direct intervention allows gradual transition,
+where autoscaling can initially be enabled only for a few PVCs for validation purposes.
+Similarly, to enable PVC autoscaling on existing seed clusters' `garden` PVCs, those will be directly annotated at
+operators' discretion.
+For details regarding the transition of existing volumes to autoscaling, see 
+[section 'Enabling Autoscaling for Existing Volumes'](#enabling-autoscaling-for-existing-volumes)
+
+![02-operator_structure.png](assets/gep29-02-operator_structure.png)
+
+_Fig.2: Operator structure_
+
+VPA is used to scale `pvc-autoscaler`. For the future security enhancement scenario, where `kube-rbac-proxy` is used
+(see [section 'Future Enhancements'](#future-enhancements)),
+that container is expected to be excluded from autoscaling, and to run with fixed resource requests. Pod evictions
+driven by that lean container are unjustified, as they would only achieve minimal resource savings, and disrupt
+the main container.
+
+The feature is deployed behind a dedicated feature gate. When disabled, `pvc-autoscaler` is removed, and related
+annotations are removed from the shoot control plane observability StatefulSets, and the seed observability StatefulSets.
+
+#### Initial volume size and MaxAllowed
```

Comment:
Because I view it as part of the deployer behaviour. The specific values are impactful enough to be worth listing, but I don't want to promote the table to a higher level. Impactful as it is, it's still a fairly low level implementation detail.

However, your comment made me reconsider the title for the parent section. Instead of "PVC Autoscaler - Deployer Structure", I think I should use just "PVC Autoscaler - Deployer", since the content of the section actually goes beyond static structure, and into behaviour/runtime effects. WDYT?